tensorboard/webapp/hparams/_redux/hparams_data_source.ts

```diff
@@ -0,0 +1,182 @@
+    const formattedExperimentIds = this.formatExperimentIds(experimentIds);
+    return this.http
+      .post<BackendHparamsExperimentResponse>(
+        `/${this.getPrefix(
+          experimentIds
+        )}/${formattedExperimentIds}/${HPARAMS_HTTP_PATH_PREFIX}/experiment`,
+        {experimentName: formattedExperimentIds},
+        {},
+        'request'
+      )
+      .pipe(
+        map((response) => {
+          return {
+            hparams: response.hparamInfos.map((hparam) => ({
+              ...hparam,
+              domain: getHparamDomain(hparam),
+            })),
+            metrics: response.metricInfos.map((info) => ({
+              ...info,
+              tag: info.name.tag,
+            })),
+          };
+        })
+      );
+  }
+
+  fetchSessionGroups(
+    experimentIds: string[],
+    hparamsAndMetricsSpecs: HparamsAndMetricsSpecs
+  ) {
+    const formattedExperimentIds = this.formatExperimentIds(experimentIds);
+
+    const colParams: BackendListSessionGroupRequest['colParams'] = [];
+
+    for (const hparam of hparamsAndMetricsSpecs.hparams) {
+      colParams.push({hparam: hparam.name});
+    }
+    for (const mectric of hparamsAndMetricsSpecs.metrics) {
+      colParams.push({
+        metric: mectric.name,
+      });
+    }
+
+    const listSessionRequestParams: BackendListSessionGroupRequest = {
+      experimentName: formattedExperimentIds,
+      allowedStatuses: [
+        RunStatus.STATUS_FAILURE,
+        RunStatus.STATUS_RUNNING,
+        RunStatus.STATUS_SUCCESS,
+        RunStatus.STATUS_UNKNOWN,
+      ],
+      colParams,
+      startIndex: 0,
+      // arbitrary large number so it does not get clipped.
+      sliceSize: 1e6,
+    };
+
+    return this.http
+      .post<BackendListSessionGroupResponse>(
+        `/${this.getPrefix(
+          experimentIds
+        )}/${formattedExperimentIds}/${HPARAMS_HTTP_PATH_PREFIX}/session_groups`,
+        listSessionRequestParams,
+        {},
+        'request'
+      )
+      .pipe(
+        map((sessionGroupsList) => {
+          const runToHparamsAndMetrics: RunToHparamsAndMetrics = {};
+
+          // Reorganize the sessionGroup/session into run to <hparams,
+          // metrics>.
+          for (const sessionGroup of sessionGroupsList.sessionGroups) {
+            const hparams: HparamValue[] = Object.entries(
+              sessionGroup.hparams
+            ).map((keyValue) => {
+              const [hparam, value] = keyValue;
+              return {name: hparam, value: value as DiscreteHparamValue};
+            });
+
+            for (const session of sessionGroup.sessions) {
+              for (const metricValue of session.metricValues) {
+                const runName = metricValue.name.group
+                  ? `${session.name}/${metricValue.name.group}`
+                  : session.name;
+                const runId = `${formattedExperimentIds}/${runName}`;
+                const hparamsAndMetrics = runToHparamsAndMetrics[runId] || {
```

Comment:
This logic was essentially copied from the `runs_data_source` and I was initially planning on solving this in a future pr. However, I have realized that it would be simplest to solve if I store a slightly remapped version of the response.

I plan to do the majority of the mapping in a selector.